auto_round/autoround.py

```diff
@@ -690,13 +705,25 @@ def forward(m, hidden_states, *positional_args, **kwargs):
                                 self.inputs[name][key].extend(list(torch.split(alibi.to("cpu"), 1, dim=0)))
                             else:
                                 self.inputs[name][key] = list(torch.split(alibi.to("cpu"), 1, dim=0))
-                    elif "position_ids" in key:
-                        if key not in self.inputs[name].keys():
-                            self.inputs[name][key] = list(torch.split(kwargs[key].to("cpu"), 1, dim=0)) \
-                                if self.not_share_position_ids_flag \
-                                else to_device(kwargs[key], device=torch.device("cpu"))
+                    elif "position_ids" in key or 'cache_position' in key:
+                        if self.train_bs == 1 and self.not_share_rotary_pos_emb_flag:
+                            if key not in self.inputs[name].keys():
+                               self.inputs[name][key] = [to_device(kwargs[key], device=torch.device("cpu"))]
+                            else:
+                                self.inputs[name][key].append(to_device(kwargs[key], device=torch.device("cpu")))
+                        elif key not in self.inputs[name].keys():
```

Comment:
The code is getting more and more complicated, which is hard to maintain

Reply:
will refine multimodal code later